documentation/sdk-breaking-changes-categories.md

```diff
@@ -0,0 +1,101 @@
+# Azure Go SDK Breaking Changes Categories
+
+## 1. Naming Changes
+
+### Model/Type Naming
+
+- **TypeSpec**: Model, enum, union name changes
```

Comment:
I'm guessing these rules are going to be duplicated all over the place. Considering these are really TypeSpec-controlled, should we move it there?